### PR TITLE
Fixes hot reload when running Windows

### PR DESCRIPTION
**Pull request by luffs**
Seems to be an issue since Windows uses the backslash character as folder separator.

### CODE REVIEW

**Comment by GianlucaGuarini** [comment]
thanks for your pull request. Can we try using [`path.normalize`](https://nodejs.org/api/path.html#path_path_normalize_path) [here](https://github.com/riot/webpack-loader/blob/master/src/index.js#L42)? This patch is still a bit too hacky for me.

File: src/index.js [hunk 1]
```diff
@@ -7,6 +7,9 @@ import {getOptions} from 'loader-utils'
  * @returns {string} the code needed to handle the riot hot reload
  */
 function hotReload(path) {
+  if (process.platform === 'win32') {
```
Comment (GianlucaGuarini) [inline src/index.js]: Please avoid platform hacks and try to use the native `path` methods like suggested here https://github.com/riot/webpack-loader/pull/26#issuecomment-489450583

**Comment by luffs** [comment]
Path.normalize doesn't solve the problem, (I've tried it). The resulting path will contain one backslash, when two are needed since it's added to a script string that will be evaluated.

> When multiple, sequential path segment separation characters are found (e.g. / on POSIX and either \ or / on Windows), they are replaced by a single instance of the platform-specific path segment separator (/ on POSIX and \ on Windows).

The if-statement could probably be removed without any determent to Linux or OS-X though.

File: src/index.js (hunk 1 above)
Comment (luffs) [reply src/index.js]: Path.normalize doesn't solve the problem, but the if-statement is probably not needed.

**Comment by GianlucaGuarini** [comment]
The question is why `resourcePath` contain these backslashes. Maybe someone from the webpack community could help us. I will try pinging the webpack repo

**Comment by GianlucaGuarini** [comment]
@luffs 
webpack version?  
Node.js version?

**Comment by luffs** [comment]
Shouldn't it? I don't think this an issue with Webpack.
Since Linux allows backslashes in file-names, somebody who (for some insane reason) names their file my\tag.riot will have the same problem. The resulting string from the function hotReload will be:

```
if (module.hot) {
    const hotReload = require('@riotjs/hot-reload').default
    module.hot.accept()
    if (module.hot.data) {
      const component = require('./my\tag.riot').default;
      hotReload(component)
    }
  }
})()`
```
... when is should be

```
if (module.hot) {
    const hotReload = require('@riotjs/hot-reload').default
    module.hot.accept()
    if (module.hot.data) {
      const component = require('./my\\tag.riot').default;
      hotReload(component)
    }
  }
})()`
```

Webpack 4.30.0
Node v10.15.3

**Comment by GianlucaGuarini** [comment]
thank you. This issue was fixed with https://github.com/riot/webpack-loader/pull/29

**Pull request closed without merging**